### DISCUSSION TITLE
Garbage collection?

### DISCUSSION THREAD

**Discussion by mmliam** [micropython/micropython#9729] [General]
I have a relatively short program that declares a few variables, performs a brief network operation then goes into deep-sleep awaiting a reboot.

Given these operations is "garbage collection" necessary; won't all memory be reinitialized at re-boot?

MM

**Comment by davefes** [comment]
Maybe robert-hh comment at the end.
[https://github.com/orgs/micropython/discussions/9345](url)
The "link" may not work.

davef

**Reply by mmliam** [reply]
Link is dead

**Reply by davefes** [reply]
I can't get them to work ...  do a copy/paste or go to ideas and look at the POWER_ON RESET idea.

**Reply by dlech** [reply]
Don't use the "Add a link" button, just write the link and GitHub will make it work:  https://github.com/orgs/micropython/discussions/9345

**Reply by davefes** [reply]
Gosh ... thanks.

**Comment by jimmo** [comment]
> Given these operations is "garbage collection" necessary; won't all memory be reinitialized at re-boot?

Yes that is correct. All MicroPython state will be re-initialised, the heap is zero-initialised at start-up.

**Reply by mmliam** [reply]
jimmo, Thanks for the confirmation
